File: C#Part2/HomeworkStringsAndTextProcessing/SubStringInText/SubStringInText.cs
//Write a program that finds how many times a sub-string is contained in a given text (perform case insensitive search).
//Example:
//The target sub-string is in
//The text is as follows: We are living in an yellow submarine. We don't have anything else. inside the submarine is very tight. So we are drinking all the day. We will move out of it in 5 days.

using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

class SubStringInText
{
    static void Main()
    {
        string input = "We are living in an yellow submarine. We don't have anything else. Inside the submarine is very tight. So we are drinking all the day. We will move out of it in 5 days.";

        input = input.ToLower();

        int count = 0;

        for (int i = 0; i < input.Length - 1; i++)
        {

            if (input.Substring(i, 2) == "in")
            {
                count++;        
            }
        }
        Console.WriteLine(count);
    }
}

File: 6 JavaScriptBasic/proverka na domashni/StringsJS/03. Sub-string in text/Sub-string in text.js
// Write a JavaScript function that finds how many times a substring is 
// contained in a given text (perform case insensitive search).

function countRepeatingSubstring(text, word) {
    var match = text.match(new RegExp(word, 'gi'))
    return match && match.length || 0
}
console.log(countRepeatingSubstring('We are living in an yellow submarine.We don\'t have anything else. Inside the submarine is very tight. So we are drinking all the day. We will move out of it in 5 days.', 'in'))
